src/main/java/pl/allegro/tech/opel/ListInstantiationExpressionNode.java

```diff
@@ -0,0 +1,33 @@
+package pl.allegro.tech.opel;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ListInstantiationExpressionNode implements OpelNode {
+    private final Optional<ArgumentsListExpressionNode> listElements;
+
+    public ListInstantiationExpressionNode(ArgumentsListExpressionNode listElements) {
+        this.listElements = Optional.of(listElements);
```

Comment:
ArgumentsListExpressionNode not allowed empty list. I add EmptyArgumentsListExpressionNode class to solve this.